tensorboard/backend/application.py

```diff
@@ -337,6 +342,55 @@ def _create_wsgi_app(self):
     app = _handling_errors(app)
     return app
 
+  @wrappers.Request.application
+  def _serve_plugin_entry(self, request):
+    """Serves a HTML for iframed plugin entry point.
+
+    Args:
+      request: The werkzeug.Request object.
+
+    Returns:
+      A werkzeug.Response object.
+    """
+    name = request.args.get('name')
+    plugins = [
+        plugin for plugin in self._plugins if plugin.plugin_name == name]
+
+    if not plugins:
+      return http_util.Respond(
+          request, 'Plugin not found', 'text/plain', code=404)
+
+    if len(plugins) > 1:
+      # Technically is not possible as plugin names are unique and is checked
+      # by the check on __init__.
+      return http_util.Respond(
+          request, 'Invariant error', 'text/plain', code=400)
+
+    plugin = plugins[0]
+    module_url = plugin.frontend_metadata().es_module_path
+    if not module_url:
+      return http_util.Respond(
+          request, 'Plugin is not module loadable', 'text/plain', code=400)
+
+    # Prevent a malicious plugin from using an absolute path that would make this trusted
```

Comment:
This is fine and good, but I’d be interested to know what attack it’s
preventing. Plugins can already put arbitrary contents into their
script, so they could just fetch the remote script in the Python code
and serve it inline. What would they gain by being able to directly load
from another domain?

Also, just a note: this doesn’t prevent directory traversal (`..`)
attacks.